pyk/src/pyk/kcfg/kcfg.py

```diff
@@ -1040,6 +1040,16 @@ def create_split(self, source_id: NodeIdLike, splits: Iterable[tuple[NodeIdLike,
         self.add_successor(split)
         return split
 
+    def create_split_by_nodes(self, source_id: NodeIdLike, target_ids: Iterable[NodeIdLike]) -> KCFG.Split | None:
+        """Create a split without crafting a CSubst."""
+        source = self.node(source_id)
+        targets = [self.node(nid) for nid in target_ids]
+        substs = [source.cterm.config.match(target.cterm.config) for target in targets]
+        if None in substs:
+            return None
+        csubsts = [CSubst(subst, target.cterm.constraints) for subst, target in zip(substs, targets, strict=True)]
+        return self.create_split(source.id, zip(target_ids, csubsts, strict=True))
```

Comment:
In the substitution-based approach of `CSubst.apply`, we directly set the `CSubst` for the split to the constraints of the target.

If `CSubst.apply` were to use a combination approach, we could set the `CSubst` for the split to `target.constraints - source.constraints`.

Reply:
We should use the combination approach, and do the delta of constraints instead. A `CSubst` applied to a `CTerm` should apply the substitution and append the constraints, so that we're guaranteed to always go to a more specific state.